src/main/java/com/feedzai/commons/sql/abstraction/ddl/DbFk.java

```diff
@@ -52,8 +59,11 @@
      * @param foreignTable   The foreign table.
      */
     private DbFk(List<String> localColumns, List<String> foreignColumns, String foreignTable) {
-        this.localColumns = localColumns;
-        this.foreignColumns = foreignColumns;
+        // Sorting to make sure that equal method run in O(n) instead of O(n^2) for comparing 2 lists.
+        Collections.sort(localColumns);
+        Collections.sort(foreignColumns);
+        this.localColumns = ImmutableList.copyOf(localColumns);
+        this.foreignColumns = ImmutableList.copyOf(foreignColumns);
```

Comment:
Should we clone first and then sort ? To avoid changing the input lists.

Reply:
I don't think that is necessary in this case. The order of the items in those lists is irrelevant.

Reply:
You can't sort these columns, at least like this.
It would be ok if this class was used only for comparison, but it also contains information to build foreign key constraints when creating a table.
Each local column should be considered a pair along with a "foreign column"
(in SQL a foreign column is actually a local column, so this designation is actually wrong ─ see #201 )

For example, to get this
`FOREIGN KEY (b, c) REFERENCES other_table (c1, c2)`
we would add `b` and `c` to `localColumns`
and `c1` and `c2` to `foreignColumns`.
Columns `b` and `c1` need to be paired: the data types in these columns have to match, and values inserted in `b` need to be present in `c1`.
Same for the pair `c` and `c2` (they may have the same data type or values as `c1` for example, but that would be just a coincidence, not a requirement).

Even keeping the order `b,c` and `c1,c2`, instead of having `c,b` and `c2,c1`, may be important, but I'm not sure about this.